.mergify.yml

```diff
@@ -8,13 +8,65 @@ pull_request_rules:
       label:
         add:
           - ceph
+
   # if there is a conflict in a backport PR, ping the author to send a proper backport PR
   - name: ping author on conflicts
     conditions:
       - conflict
     actions:
       comment:
         message: This pull request has merge conflicts that must be resolved before it can be merged. @{{author}} please rebase it. https://rook.io/docs/rook/master/development-flow.html#updating-your-fork
+
+  # automerge on master only under certain strict conditions
+  - name: automerge merge master with specific label and approvals with code change
+    conditions:
+      - author=mergify[bot]
+      - base=master
+      - "#approved-reviews-by>=1"
+      - "#changes-requested-reviews-by=0"
+      - label!=do-not-merge
+      - label=ready-to-merge
```

Comment:
doubts: 
1) approved reviewers after approving need to add label `ready-to-merge` to get it merged?
2) not sure if any of these conditions ``` "#review-requested=0", "#changes-requested-reviews-by=0", "#commented-reviews-by=0```" needed.

Reply:
> doubts:
> 
> 1. approved reviewers after approving need to add label `ready-to-merge` to get it merged?

Yes. So basically, you are done reviewing, you approve, you think twice, then set the label, go do something else, and once CI is green it will get merged. This extra step to set the label is more of a security.

> 2. not sure if any of these conditions ` "#review-requested=0", "#changes-requested-reviews-by=0", "#commented-reviews-by=0`" needed.

I don't think we need those.

Reply:
Thanks for explaining.